### ISSUE TITLE
Discussion: Replace word "Column" with "Layer"?

### ISSUE THREAD

**Issue by manongjohn** [opentoonz/opentoonz#1406]
With the introduction of the Timeline layout of the Xsheet, we have the potential for a bit of confusion regarding the use of the word "Column" throughout the application. (it actually happened in a PR discussion)

Prior to the Timeline, we referenced items in the Xsheet as being in Columns/Frames. Timeline mode is the Xsheet flipped that on it's side so technically it should be referenced as Rows/Frames.

Currently, context menus, hints and other verbiage still say "Column" which would be confusing in Timeline mode, because it's no longer in a column. It's now a row.

Should we do the following?
1) Change "Column" to "Layer".
2) Change default column names from "Col#" to "Layer#"

Consider even the Toonz Harlequin documentation describes the Xsheet columns as layers and we already call Xsheet rows as "frames":

"The xsheet, the digital version of the traditional exposure sheet, allows you to control the timing of all of the elements of a scene. It is organized in columns and rows: **each column contains a layer of the animation** and **each row represents a frame contents**. "

Plus, people coming from other animation and even drawing software are more likely used to the term "Layer".

We could just leave it as "Column" in Xsheet mode, and "Layer" in Timeline mode, but would need to put in logic to switch terminology based on orientation everywhere it is needed.  I think we should avoid that if at all possible.

Thoughts? Suggestions?

**Comment by RodneyBaker** [comment]
This is a duplicate of another report but I can't seem to find it.  It may be part of a larger report such as relates to Xsheet or Timeline development.  The discussion was ongoing with nothing being resolved but everyone open to more discussion.  The consensus appeared to be that the term 'Layer' would be used in the Timeline but the Xsheet left as is currently.

Thanks for the reference to the word 'layer' being used in the Toonz Harlequinn documentation.

Recommend close with a reference to the duplicated issue (when it is located).

**Comment by manongjohn** [comment]
We were talking about this in PR #1376, but wanted to separate it from that PR since it is outside that scope.

**Comment by manongjohn** [comment]
Sorry for the confusion.  I should have referenced the PR.  The PR hasn't been merged yet so it can't be closed yet.

**Comment by gab3d** [comment]
@manongjohn, I agree with you on changing the name (of xsheet columns/timeline rows) to "Layer" and the name of new ones, accordingly, to "Layer#". 
Is what seems more consistent (it resists well the dual nature of the xsheet/timeline concept without alteration), and has also the benefit of being a more "universally known" concept to new users.

**Comment by openanim** [comment]
@manongjohn  An option like you have done in another of your jobs might be a solution?

**Comment by manongjohn** [comment]
@openanim That won't work in this situation.  We actually want everyone using the same terminology to avoid confusion.

**Comment by Rayek** [comment]
Xsheets traditionally have "Columns", not "Layers". I dislike going against a naming convention that was established as standard over 60 years ago in the animation industry. 

A big NO from me - we should **NOT** change "Column" to "Layer" when talking about an Xsheet.

Having said this, I completely agree a horizontal timeline should not be using "columns" when the industry standard is obviously a "layer" as in "stacked layers". 

In short, horizontal timelines have "layers", while Xsheets have "Columns". I am aware this is down to semantics, but we should be keeping to the respective industry standards.

**Comment by manongjohn** [comment]
@Rayek True.  Changing terminology of the Xsheet column is probably a bad idea in the fact that the industry Xsheet has always been called columns and won't change.  Calling it otherwise just for this application is just going to make our application look odd.

I just had a thought. Most of this is just label changing, thankfully, so perhaps we should just do something like this?

Select All Keys in this **Column** -> Select All Keys in this **Xsheet Column/Timeline Layer**

It will maintain the Column/Layer terminology where appropriate without having to change any code.  It may look a little ugly but it should avoid confusion and people will understand the references.

**Comment by Rayek** [comment]
@manongjohn I actually wouldn't mind that solution at all.

**Comment by gab3d** [comment]
@Rayek OK, I actually don't minded replacing the "term" column by "layer" in xsheets, because I come from using paint/compositing programs, and not really used the traditional workflow with xsheets before. But I can totally understand that if the term "column" is a solidly established term while working with xsheets, it makes sense to preserve it.

@manongjohn could it be something shorter like: 
Select All Keys in this Column (Layer)
...or something in that style? I think people will understand it anyway.

**Comment by manongjohn** [comment]
I believe this issue can be closed.

The answer to my question is  "No" in order to keep the Xsheet industry standard terminology intact.

Looking briefly at possibly changing just the Timeline default names to "Layer#", it's a bit problematic since there are numerous places where the default names are created as "Col#" outside any Xsheet window (i.e schematic view).  In those cases we may not know what orientation should be used and would default to "Col#" anyway.

The least we can do is modify the some references to be "Column/Layer",  "Column (Layer)" or something like that but if the default names are not changed to "Layer#", it wouldn't make much sense.

I think it's best, for now at least, to leave it as Columns/Frames, regardless of orientation.

Thanks all for the feedback. Curiosity satisfied. :)

**Comment by turtletooth** [comment]
Thank you @manongjohn for looking into this.  Thank you all for your perspectives.

**Comment by gab3d** [comment]
I've checked what ToonBoom does, when creating new layers/columns (as they
have the same problem as we do), and they call them Drawing1, Drawing2,
etc. which, even it doesn't convince me, it shows us two thigns:

1. they looked for a single, orientation neutral, term, versus one for each
orienttion.

2. they chose something different from "Column" and nobody in the industry
is diying for that. :D

**Comment by RodneyBaker** [comment]
Perhaps we can create a new term:  Pflnp (TM)  

As in, "Please swap the first and second pflnps so the first appears in front of the second."

Then as other software begin to adopt the term OpenToonz can charge a royalty for its usage.

Problem solved.  :)

**Comment by manongjohn** [comment]
@gab3d Layer makes a lot more sense than Drawing.  Xsheet is more than just drawings to me.  I wonder what their discussion was like trying to figure out what wording to use.  I'd really like to understand why they chose Drawing when most applications use Layer.  Apparently they threw the idea standard terminology out the window. How dare they!

I still think we should change everything to Layer. It will still be referred to as Column in Xsheet mode simply because it's in an an actual column. It's just the label will say Layer. I'd only make that change if the powers-that-be agreed to it otherwise no point making the changes.  I'm sure if it was changed and merged in, some parts of the community may grumble but as with anything, get used to it.  I've yet to see anyone stop using something just because they changed the wording on something. :) 

As the timeline isn't out to everyone yet and only being used by a handful of people, I guess we'll see what the feedback is after the next official release. 

@RodneyBaker LOL.

**Comment by gab3d** [comment]
@manongjohn, also to me "Layer" would be a better name, and above all, more understandable to new users that haven't ever used an xsheet.

Another thing I think should be changed is the stacking order in the timeline. Layers above others in the timeline should appear in front of those ones in the screen.
I mean, the whole point of having a Timeline in OT is to please users who are used to work with timelines, but ...we're giving them a timeline that works in an inverse fashion. why?!

**Comment by manongjohn** [comment]
@gab3d The original developer of the timeline did it this way, probably because he was just trying to get a horizontal version of the xsheet working.  It was quite a lot of work and did a tremendous job. He was almost done but then decided to no longer work on it so I took over the development to finish what he started.

My next major focus is to reverse the layering of the timeline to be bottom up, to keep in line with other software.  I hope to get it done by the next official release, whenever that is but not sure I'll make it at this rate.

I had started investigating  what it would take to flip it and was able to flip it to a degree, but then discovered there was a significant number of other changes that needed to be made for it to all work right.  Additionally, there is some enhancement work that was done by the original developer that was based on the timeline being introduced. Unfortunately the timeline code in the new project diverged greatly so I can't easily merge the original timeline PR into it. And that project introduced some code that would make it a bit easier for me to flip the timeline, I think.

So at the moment, I'm trying to merge the timeline code that was introduced into the enhancement project but it's extremely painful because the code changed so much.  Soon as that is done, I will work on flipping the timeline.

**Comment by gab3d** [comment]
@manongjohn, oh I see. Thanks for pointing this out!
I'm just happy to know that the issue is being taking care of, no matter how fast or not this finally is :)

**Comment by artisteacher** [comment]
@RodneyBaker Pflnp is genius but might be hard to pronounce. What about "Colulay"? 😉

One of the things I think is a strength of OT is the focus on traditional animation workflows and from that side, column still makes sense. I've been working on updating handouts for my students and writing column/layer everywhere is not ideal. Having a single term would be easier for new users. 

A compromise might be a preference option to use layer instead of column.

In the meantime, I'm hoping this graphic will make the terminology clearer for my students:
![xsheet-timeline](https://user-images.githubusercontent.com/21159116/29374585-f216663c-827f-11e7-8edd-a8825ca9de66.png)

**Comment by gab3d** [comment]
@artisteacher, mmm... not sure if the last line convince me.
a cel is basically a drawing, but is not a "frame" per se, a frame could contain cells or be empty. So in that sense, a frame is kind of a container for a cel, regarding time. Much in the same way a layer/column is a container for a cel, regarding space.

So in your last line of the graphic I'd put:
Layer/Column 1 in Frame 1 + Layer/Column 2 in Frame 2 + ..... = Composited Frame 1 / Final Frame 1 / Output Frame 1

**Comment by RodneyBaker** [comment]
@artistteacher   I like your graphic!
Like @gab3d I got stuck on that last line at the use of 'cell' and frame although not quite for the reasons he did.  I think if you would have said 'on Cell' it might more accurately capture traditional approach.  It's purely by accident but but an interesting connection that 'cel' and 'cell' almost merge in this approach.

For everyone , this might be useful to think of this switching back and forth in terms of multiplane camera with the horizontal (the original form of cell placement (largely ushered in by Charles Fleische) or as  vertical (as developed by Disney, who used both but is best known for the upright vertical multiplane setup).  The frames were held in place at intervals corresponding to our Columns in an xsheet with the changing out of cels on each frame and the click of a camera capturing what is 'in frame'.  This setup is basically the same whether the setup is horizontal or vertical.  

It is interesting to note that the multiiplane cameras 'planes' were actual upright frames of wood or other similar material that could be moved closer or farther away from the camera.  This is what is captured in the instructions on the exposure sheet and managed by directing which level of art captured on a cell is dictated to be in any given column.

In digital terms we do tend to think of data being 'in a cell' versus on it but I hope you see the point.

It's also interesting that while there were many innovations that were being crafted to improve production one of the innovations Walt Disney patented was for this multiplane setup to achieve automatic shadows.  See [US patent 2201689](https://www.google.com/patents/US2201689) for more information on that aspect.

**Comment by gab3d** [comment]
@Rodney, the concept of "plane" is what I'd like to pick from your very
interesting post. :)

plane = layer = column

**Comment by RodneyBaker** [comment]
Fun stuff!

I'm not sure the term plane quite fits and I'll endeavor to explore why.

The term plane is certainly a fundamental one and it can be found in both vertical or horizontal orientations.  Of course without the third dimension of depth it is more likely to be perceived as a line than a plane.  A plane can be considered a straight line with depth.  If freed from any two of it's planar axes then that plane can orient and align itself in a third dimension space.  

With regard to OpenToonz, the majority of our interactions with it are confined to the X and Y axes so we basically have the Camera Plane as a construct and an illusion of perspective and depth as defined on imaginary planes in front of that camera plane that symbolize/suggest such things as Ground plane*, Horizon, etc. but note that those are actually lines we perceive as defining various planes and not actually planes.  I do suppose that any contrived (read:  imaginary) flat surface running parallel to a Camera plane could rightly be referred to as a plane.   

To finish this basic theme...
The fourth dimension then is one of 'states', namely that of the capturing of perceptive moments (or percieved movement) in a time frame (!) from two or more given points of measure.  We can then define such measurement as having various extents and therefore of falling inside or outside of specific ranges.

*Just as the Camera plane is extrapolated from the position of the camera I will guess that the Ground plane and other planes could be extrapolated as well so I'm not quite being accurate in my exploration.  But all these elements could be considered fundamentally pointers to constructs in time and space.  This is where we tie in with the channels of the function editor which In OpenToonz provides another view of 2D time and space and 3D programs use for displaying a 2D view of changes occurring in three dimensional space.

**Comment by RodneyBaker** [comment]
I should have just linked to this video where Walt explains the use of their Multiplane camera.
I assume that most of you have already seen it but its a fun video to watch again.

https://www.youtube.com/watch?v=8d4-AUwkKAw

**Comment by artisteacher** [comment]
The Harlequin manual describes levels as being "exposed in cells", so I based this off of that terminology, which seems to be the spreadsheet point of view. In a spreadsheet, content is "in" the cell. I'm not sure about using "cel", as I've used the term drawing in most of my handouts (except for the cel animation one), again based on the manual. Cel is definitely more traditional but drawing might be more universal, particularly when dealing with a variety of workflows.  I do want to make sure this is as accurate and clear as possible but it is tricky with the dueling spreadsheet/animation terminology. For now I might go with @gab3d's 1st suggestion to keep it simple.

Thanks @RodneyBaker and @gab3d for the clarifications.

**Comment by RodneyBaker** [comment]
No, I wouldn't use 'cel' as that is a direct reference to the plastic celluloid sheets that are very very rarely used any more.  We have to use 'cell' where appropriate.  I just think it's neat with those two terms seem to be trying to reconverge.

I know it may be confusing but I don't see that any entirely a bad thing.
It's good for students to get a view for how technology changes how we produce animation.
Technology is aways a driver of change.
